hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java

```diff
@@ -130,7 +129,9 @@ public void submit(@NonNull final TransactionBody txBody, @NonNull final Bytes t
         // FUTURE This should be deprecated and removed. We do not want this in our production system.
         if (txBody.hasUncheckedSubmit()) {
             // We do NOT allow this call in production!
-            if (isProduction) {
+            // check profile dynamically, this way we allow profile overriding in Hapi tests
+            final var hederaConfig = configProvider.getConfiguration().getConfigData(HederaConfig.class);
+            if (Profile.valueOf(hederaConfig.activeProfile()) == Profile.PROD) {
```

Comment:
Interesting. I think `HederaConfig.activeProfile()` should return a `Profile` directly. Wonder why it returns a `String`. 🤔

Reply:
I'm not sure why it was done like this. 
However I made it to work with `Profile`.